dlp/snippets/inspect/inspect_bigquery_with_sampling.go

```diff
@@ -0,0 +1,185 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package inspect
+
+// [START dlp_inspect_bigquery_with_sampling]
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+	"cloud.google.com/go/pubsub"
+)
+
+// inspectBigQueryTableWithSampling inspect bigQueries for sensitive data with sampling
+func inspectBigQueryTableWithSampling(w io.Writer, projectID, topicID, subscriptionID string) error {
+	// projectId := "your-project-id"
+	// topicID := "your-pubsub-topic-id" or provide a topicID name to create one
+	// subscriptionID := "your-pubsub-topic-id" or provide a topicID name to create one
+
+	ctx := context.Background()
+
+	// Initialize a client once and reuse it to send multiple requests. Clients
+	// are safe to use across goroutines. When the client is no longer needed,
+	// call the Close method to cleanup its resources.
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify the BigQuery table to be inspected.
+	tableReference := &dlppb.BigQueryTable{
+		ProjectId: "bigquery-public-data",
+		DatasetId: "usa_names",
+		TableId:   "usa_1910_current",
+	}
+
+	bigQueryOptions := &dlppb.BigQueryOptions{
+		TableReference: tableReference,
+		RowsLimit:      int64(10000),
+		SampleMethod:   dlppb.BigQueryOptions_RANDOM_START,
+		IdentifyingFields: []*dlppb.FieldId{
+			{Name: "name"},
+		},
+	}
+
+	// provide storage config with bigqueryOptions
+	storageConfig := &dlppb.StorageConfig{
+		Type: &dlppb.StorageConfig_BigQueryOptions{
+			BigQueryOptions: bigQueryOptions,
+		},
+	}
+
+	// Specify the type of info the inspection will look for.
+	// See https://cloud.google.com/dlp/docs/infotypes-reference for complete list of info types
+	infoTypes := []*dlppb.InfoType{
+		{Name: "PERSON_NAME"},
+	}
+
+	// Specify how the content should be inspected.
+	inspectConfig := &dlppb.InspectConfig{
+		InfoTypes:    infoTypes,
+		IncludeQuote: true,
+	}
+
+	// Create a PubSub Client used to listen for when the inspect job finishes.
+	pubsubClient, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		return err
+	}
+	defer pubsubClient.Close()
+
+	// Create a PubSub subscription we can use to listen for messages.
+	// Create the Topic if it doesn't exist.
+	t := pubsubClient.Topic(topicID)
+	if exists, err := t.Exists(ctx); err != nil {
```

Comment:
This if/else logic for topic and subscription seems to clutter a snippet that is already long. Up to your discretion to keep or remove but thought I'd mention it.

Reply:
- Initially, I had also this thought but then I thought that this snippet will be helpful to the user because they don't need to bother about creating a pub-sub subscription and topic so it takes away the burden of the user, that's why I have kept it, but let me know what should be the ideal case in this scenario and should we need to follow that or not. Thanks again @tritone

Reply:
IMO better to keep branching logic out of the snippet as much as possible -- either create the resources in the sample or tell the user they must be created before hand. But ultimately I think this is up to the discretion of the dlp team so you and @annadata can feel free to make the call.

Reply:
@tritone Thanks for the feedback.
This approach allows us to run the sample without having to pre-create topic/subscription and without without worrying about cleanups.